### DISCUSSION TITLE
Re: Project structure, assembly/package naming, etc

### DISCUSSION THREAD

**Discussion by mwpowellhtx** [OrchardCMS/OrchardCore.Commerce#103] [Ideas]
Diving in and getting myself oriented to the project, repo, etc. Suggest structuring the solution:

```
docs/
src/
```

With the _VS Solution_ in the `src/` directory, allowing for top level repo directories, supporting `docs/`, etc.

Also, I gather has not been published yet. Except possibly to local _NuGet_ sources. I think the projects `AddressDataType`, `MoneyDataType`, etc. I get why they might be separate assemblies, but if they are intent to support _OC_, _Commerce_, probably should be prefixed as such. Else if they can truly exist apart from either or both, then maybe prefix to avoid possible package conflicts. 🧠 Your thoughts?

```
OrchardCore.Commerce.DataTypes.Shipping
OrchardCore.Commerce.DataTypes.Money
```

And with both landing in a similar namespace aligned with the core project, i.e. along these lines:

```csharp
namespace OrchardCore.Commerce.DataTypes.Shipping
{
  public class Address { ... }
}

namespace OrchardCore.Commerce.DataTypes.Money
{
  public class Amount { ... }
  public class Currency { ... }
}
```

Happy to take a lead on that one. Cheers 🍻

**Comment by bleroy** [comment]
Thanks for the suggestion. Those projects were separated precisely so they can be rid of all Orchard dependencies and should remain so even in naming. I'll also pass on repo structure, as I prefer to have the projects be top-level and then have relevant structure underneath those.

**Reply by mwpowellhtx** [reply]
I'm not sure what you mean _"rid of all Orchard dependencies"_. Isn't it somewhat the point this is for use with _OC CMS_?

Also not sure what you mean _"relevant structure underneath"_. I'm suggesting moving those files and directories under a `src/`. Which allows for supporting assets, `docs/` and so forth.

In terms of structure, i.e.

```
docs/
src/
    ├ AddressDataType/
    ├ MoneyDataType/
    ├ OrchardCore.Commerce/
    ├ OrchardCore.Commerce.Tests/
    ├ SampleWebApp/
    ├ Nuget.config
    └ OrchardCore.Commerce.sln
```

But I'd also be concerned that there wasn't a `NuGet` package conflict for something as generic as `AddressDataType` or `MoneyDataType`.

Relevant structure would be preserved under `src/` directory.

**Comment by bleroy** [comment]
No, those specific libraries are made to be used primarily with Orchard core *Commerce*, but to also be usable on their own.

Nuget package name is orthogonal to namespace, that's fine. Namespaces can also be aliased if necessary. There is no issue there.

I don't see the value in changing the project structure, I'm fine with the current structure.

**Comment by bleroy** [comment]
Also, Orchard projects maintain their documentation on a separate project that then gets published to ReadTheDocs. See https://github.com/OrchardCMS/OrchardDoc and https://docs.orchardcore.net/en/latest/.

Currently, the commerce project has none.

**Reply by mwpowellhtx** [reply]
Thinking more along the lines of internal docs than public facing, published APIs, examples, etc, or what have you. However, I see.

> Nuget package name is orthogonal to namespace...

Sometimes, frequently yes, but not necessarily.

Anyway I appreciate the feedback, thank you.
